core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -18,6 +18,7 @@
 #define KOKKOS_IMPL_PUBLIC_INCLUDE
 #endif
 
+#include <Kokkos_Core.hpp>
```

Comment:
@ I gonna approve this for now but we need to fix this properly.

Reply:
And properly could mean removing the other includes below, or actually figuring out the minimal set we need.

Reply:
Yeah, this was the "heavy hammer" quick fix (including `Kokkos_HIP_UniqueToken.hpp` introduced new problems)